theseus/geometry/functorch.py

```diff
@@ -0,0 +1,52 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import threading
+from typing import Any
+
+
+class _FunctorchContext(object):
```

Comment:
Is `functorch.py` only used in `geometry` or should we move this to `core`?

Reply:
I guess since this context is being renamed as checks for groups so this place is fine. We should do the renaming in this PR though.

Reply:
There are two PRs for functorch. I plan to create a new PR after #266 and #268 are merged.

Reply:
Refactored.